wallet/src/Pos/Wallet/Web/Tracking/Sync.hs

```diff
@@ -247,10 +244,14 @@ syncWalletWithGStateUnsafe encSK wTipHeader gstateH = setLogger $ do
             trackingRollbackTxs encSK allAddresses mDiff blkHeaderTs $
             zip3 (gbTxs b) (undoTx u) (repeat $ getBlockHeader b)
 
-        applyBlock :: [CWAddressMeta] -> Blund ssc -> m CAccModifier
-        applyBlock allAddresses (b, u) = pure $
-            trackingApplyTxs encSK allAddresses mDiff blkHeaderTs ptxBlkInfo $
-            zip3 (gbTxs b) (undoTx u) (repeat $ getBlockHeader b)
+        applyBlock :: [CWAddressMeta] -> HeaderHash -> m (Maybe (CAccModifier,Block ssc))
+        applyBlock allAddresses hh = do
+            blundM <- DB.blkGetBlund hh
+            pure $ blundM <&> \(b,u) ->
+                let cam = trackingApplyTxs
+                              encSK allAddresses mDiff blkHeaderTs ptxBlkInfo
+                              (zip3 (gbTxs b) (undoTx u) (repeat $ getBlockHeader b))
+                in (cam,b)
```

Comment:
what the heck? why do you need block `b`, why not to return only `cam`? I was looking at the code about 10 mins and didn't get why it's needed, however, if it's needed, are you sure it's not enough to return only block header? block takes more space in RAM.